### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by edugfilho**

### CODE REVIEW

File: src/main/java/com/ciandt/techgallery/authorization/AuthorizeServlet.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * Gets the current user and their credentials 
+   */
+  @Override
+  public void doGet(HttpServletRequest req, javax.servlet.http.HttpServletResponse resp) throws ServletException ,IOException {
+    AuthorizationCodeFlow authFlow = initializeFlow();
+    Credential credential = authFlow.loadCredential(getUserId(req));
+    //DO STUFF HERE
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/authorization/AuthorizeServlet.java]: What stuff?

File: src/main/java/com/ciandt/techgallery/authorization/OAuth2Callback.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    resp.sendRedirect("/");
+  }
+
+  @Override
+  protected void onError(HttpServletRequest req, HttpServletResponse resp,
+      AuthorizationCodeResponseUrl errorResponse) throws ServletException, IOException {
+    resp.getWriter().print(
+        "<h3>É preciso autorizar o uso do Google+ para ter acesso ao Tech Gallery</h3>");
+    resp.setStatus(200);
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/authorization/OAuth2Callback.java]: The HTTP responde code should be 403 Forbidden in this case.

File: src/main/java/com/ciandt/techgallery/authorization/OAuthUtils.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.appengine.api.datastore.Entity;
+
+
+public class OAuthUtils {
+
+  // Client secret must be stored as a resource and included in the build.
+  // Can never be committed along with code
+  private static final String CLIENTSECRETS = "client_secrets.json";// "edu-tech-gallery-secret.json";
```
Comment (larruda) [inline src/main/java/com/ciandt/techgallery/authorization/OAuthUtils.java]: Please remove the comment of the other json file.

File: src/main/webapp/WEB-INF/appengine-web.xml [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-        <property name="java.util.logging.config.file" value="WEB-INF/logging.properties" />
-    </system-properties>
-    <static-files>
-        <exclude path="/node/**" />
-        <exclude path="/node_modules/**" />
-    </static-files>
+<?xml version="1.0" encoding="utf-8"?>
+<appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
+    <application>tech-gallery</application>
+    <version>authtest</version>
```
Comment (larruda) [inline src/main/webapp/WEB-INF/appengine-web.xml]: Please don't commit the version hardcoded.

File: src/main/webapp/.gitignore [hunk 5]
```diff
@@ -1,3 +1,2 @@
-/node/
-/node_modules/
-/lib/
+/node/
```
Comment (larruda) [inline src/main/webapp/.gitignore]: Undo this whole file.

File: src/main/webapp/js/directives.js [hunk 6]
```diff
@@ -23,4 +51,5 @@ angular.module('techGallery.directives', [])
   return function(scope, elm, attrs) {
     elm.text(version);
   };
```
Comment (larruda) [inline src/main/webapp/js/directives.js]: Resolve the conflict before comitting.

File: src/main/java/com/ciandt/techgallery/authorization/AuthorizeServlet.java (hunk 1 above)
Comment (edugfilho) [inline src/main/java/com/ciandt/techgallery/authorization/AuthorizeServlet.java]: My bad. It was meant to tell a colleague where to put his stuff, as he is working on a similar functionality.

**Pull request merged**